utils.py

```diff
@@ -0,0 +1,161 @@
+
+            _log.info(f"Assertion: Band '{output_band_name}' and statistic '{stat}'")
+            output_stat = output_band_stats[stat]
+            assert output_stat == pytest.approx(gt_value, rel=tolerance)
+
+def calculate_band_statistics(hypercube: xr.Dataset) -> dict:
+    """
+    Calculate statistics for each variable in the output cube.
+    
+    Parameters:
+        hypercube (xarray.Dataset): Input hypercube obtained through opening a netCDF file using xarray.Dataset.
+        
+    Returns:
+        dict: A dictionary containing statistics for each variable in the output cube. Keys are variable names 
+              (matching the pattern 'B01, B02, ...') and values are dictionaries containing mean, variance, min, max, 
+              and quantile statistics.
+    """
+    statistics = {}
+    band_names = [band_name for band_name in hypercube.data_vars if band_name != 'crs']
+
+    for band_name in band_names:
+        band_data = hypercube[band_name]
+        mean_value = float(band_data.mean())
+        variance_value = float(band_data.var())
+        min_value = float(band_data.min())
+        max_value = float(band_data.max())
+        quantiles = band_data.quantile([0.25, 0.5, 0.75]).values
+        
+        statistics[band_name] = {
+            'mean': mean_value,
+            'variance': variance_value,
+            'min': min_value,
+            'max': max_value,
+            'quantile25': quantiles[0],
+            'quantile50': quantiles[1],
+            'quantile75': quantiles[2],
+        }
+
+    return statistics
+
+def execute_and_assert(cube: openeo.DataCube, 
+                       output_path: Union[str, Path], 
+                       scenario_name: str,
+                       tolerance: float = 0.01) -> None:
+    """
+    Execute the provided OpenEO cube, save the result to the output path, 
+    and assert its statistics against the reference data.
+
+    Parameters:
+        cube (openeo.datacube.DataCube): The OpenEO data cube to execute.
+        output_path (Union[str, Path]): The path where the output should be saved.
+        scenario_name (str): A name identifying the scenario for reference data.
+
+    Returns:
+        None
+
+    Raises:
+        RuntimeError: If there is an issue during execution, file saving, or assertion.
+    """
+    try:
+        cube.execute_batch(output_path,
+                            title=scenario_name,
+                            description='benchmarking-creo', 
+                            job_options={'driver-memory': '1g'}
+                            )
+
+        output_cube = xr.open_dataset(output_path)
+        output_band_stats = calculate_band_statistics(output_cube)
+        groundtruth_band_stats = extract_reference_band_statistics(scenario_name)
+        assert_band_statistics(output_band_stats, groundtruth_band_stats, tolerance)
+    except Exception as e:
+        _log.error(f"Error during execution and assertion: {e}")
+        raise
```

Comment:
I don't think it's necessary to catch an exception, log an error and throw it again. Pytest should do the right thing already, you  don't need this try-except boilerplate typically